### DISCUSSION TITLE
Gitea automerge fails because of branchStatus "yellow"

### DISCUSSION THREAD

**Discussion by malheleco** [renovatebot/renovate#15853] [Request Help] [answered]
Hi everyone,
I'm currently trying to setup renovate with the code (and a maven pom.xml) in a gitea repository.
I'm having issues with the automerge feature which I have configured like so
```
{
      matchUpdateTypes: ["patch"],
      matchCurrentVersion: "!/^0/",
      automerge: true
    }
```
The PR shows that automerge is enabled, renovate also seems to check and would "want" to automerge. I still end up getting "no automerge" although I can see in the gitea frontend that the latest commit ran fine on Jenkins (tests are present) and I also approved the PR (which is opened by another that has also has write permissions and could merge the branch/PR.

I set the logs to `TRACE` level and found that it calls `"https://[gitea-server]/api/v1/repos/[ORG]/[repo-name]/commits/renovate%2F[dependency-name-here]-5.x/statuses` which returns an empty status array, thus explaining why the PR won't be automerged.
However, I'm not sure if that is a valid use of the Gitea API or if that's a bug in the API. If I get a commit hash in that branch and manually use it in my browser instead of the branch name (`renovate%2F[dependency-name-here]-5.x`), I will get a non-empty status array.
I'm running gitea 1.16.8.

I'm wondering: is this an issue I should report to the gitea project, i.e. is this a problem in their API or is this a bug in renovate?

**Comment by rarkins** [comment, marked as answer]
Please find and read their API reference for this endpoint.

**Reply by malheleco** [reply]
The [API documentation](https://try.gitea.io/api/swagger#/repository/repoListStatusesByRef) would seem to indicate that it is an issue with the gitea API. I raised the discussion here, because I wanted to be sure that it wasn't a "misuse" by renovate or maybe something that worked in the past and was (recently) changed in renovate.

**Reply by viceice** [reply]
I've no issues on my gitea. latest automerge was ~16h ago. using Jenkins and Drone as CI

**Reply by malheleco** [reply]
@viceice Thanks, then I'll try and contact my gitea "provider" to find out if the gitea API on my instance is broken. Because when I query the AP i with a branch or tag, I get no results but a commit hash works fine.

**Comment by malheleco** [comment]
I checked with the public gitea demo API and it worked as expected. So the problem seems to be specific to the gitea setup I'm working with, so I will address it there.
Sorry for bothering you and thanks for the quick replies!

**Comment by bjuergens** [comment]
>  Because when I query the AP i with a branch or tag, I get no results but a commit hash works fine.

I had the same problem. I also noticed, that manual branches (without slashes in them) did give me results via the API. Also I suspect the underlying problem is similar to the emoji-thing (https://github.com/renovatebot/renovate/issues/10264#issuecomment-853746484), and here might also be some gitea-mysql-encoding error in the background, which is silently ignored. 

My solution was simply to change the branch-prefix from "renovate/" to "renovate_" and then closing all existing branches PRs to forces them to be recreated. 

On a side note: Be sure to have branch-protection in place, because otherwise you may see some premature merges (see https://github.com/renovatebot/renovate/issues/2948)

**Reply by malheleco** [reply]
Yeah, I figured out the same yesterday @bjuergens and configured the same "workaround". I also set branchPrefixOld as recommended in the renovate docs.
Thanks for pointing it out here.
